Add new benchmark: Catalan bench

zxcvuser opened the pull request:
CatalanBench is a benchmark for tasks in Catalan that cover several evaluation areas. The datasets consist of professional translations of relevant English datasets and newly created datasets in Catalan. The README.md contains detailed information on all the tasks included in the benchmark.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2154) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2154) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2154) it.</sub>

zxcvuser commented on lm_eval/tasks/catalan_bench/paws_ca.yaml:
```diff
@@ -0,0 +1,20 @@
+validation_split: validation
+test_split: test
+process_docs: !function utils.process_docs_paraphrases
+doc_to_text: ''
```
Could you please give us more details on what you mean here? The examples provided are in English but our task is in Catalan, and looking at the outputs in our tests we can not detect any problem in the two possible continuations.

baberabb replied:
I meant to say Claude __translated__. The continuation sentences sounded to me like they were missing a preceding statement. 
Also as the context (doc_to_text) is empty, we always prepend a EOS or BOS [token](https://github.com/EleutherAI/lm-evaluation-harness/blob/main/lm_eval/api/model.py#L367). But as long as it’s fine at your end. :)

zxcvuser replied:
Oh that's interesting. The models we've tested have not shown that behavior. The prompt and thus how the sentence is "built" was taken from the existing pawsx implementation in Harness and seems to work fine for us.
Thanks for pointing the doc_to_text behavior. Again, we just followed the existing implementation to keep things parallel across languages. There should be no issue adding those tokens, unless you have another experience with this!

zxcvuser commented:
These are the changes done:
- Added the benchmark info in lm_eval/tasks/README.md
- Replaced "-" by "_" in the create_files script in flores_ca and added weight_by_size: false
- Run linters
- Remove grouping in mgsm, paws, and mgsm tasks (they were pointing to pre-existing benchmarks)
- Changed "group" to "tag" in phrases_va, arc_ca and cabreu
- Removed an if condition that was not being entered in any case in utils.py
- Removed a duplicated "test_split: true" in xquad_ca

With these, it should all be fine now. Thank you!

haileyschoelkopf reviewed (changes requested):
Hi @zxcvuser , one last change before we can merge: would you be able to run the precommit/linter against `create_yamls_flores_ca.py` and fix the errors it raises? Thanks!

zxcvuser commented:
> Hi @zxcvuser , one last change before we can merge: would you be able to run the precommit/linter against `create_yamls_flores_ca.py` and fix the errors it raises? Thanks!

Done! Thank you
